### DISCUSSION TITLE
How _unique_ should the machine ID be?

### DISCUSSION THREAD

**Discussion by tukusejssirs** [statelyai/xstate#3018] [Q&A] [answered]
I’ve read in the docs that the machine ID should be unique. My question is how unique should it be?

I want to use XState to monitor the state of CNC machines. Actually, there are multipe features we monitor on the machines (light indicator, production, alarms, …) and I think I’d like to use a state machine per feature per machine.

Now I use Nest.js with TypeScript and each feature is a separate module, therefore I think I want to use the CNC machine IDs as state machine IDs. Would it make XState unhappy if I use the same state machine ID in different modules? Do I really need to prepend/append a string (e.g. the feature name)?

**Comment by davidkpiano** [comment, marked as answer]
The uniqueness of the machine ID is most important for dev tooling, as that is how the interpreted machine will show up on a list of actors. But for all intents and purposes, you can consider it the machine's "name".

If you use different machines with the same ID, they will show up in dev tools like `someId (x:14)` and `someId (x:26)`, so they are still "different" but with the same name.

**Reply by tukusejssirs** [reply]
Thanks!

I was thinking about have a `private` service (class) level array property with all the state machine objects. Is it this necessary or can I get a machine using its ID at runtime? Something like `xstate.getmachine(id)`?

**Reply by eavestn** [reply]
How do I update the docs to include this information? `id` - IMO - is generally considered to be a uniquely identifying attribute of an entity.

**Reply by davidkpiano** [reply]
> How do I update the docs to include this information? `id` - IMO - is generally considered to be a uniquely identifying attribute of an entity.

You can make a PR to the Stately docs here: https://github.com/statelyai/docs
